### PR TITLE
New website. Closes #5175 

### LINKED ISSUE #5175: Would you be interested in a new homepage design?
### Description of feature

Hi! This is more so a feature offer than a feature request. I believe that Dokku is much more powerful and has a lot more to offer than the current homepage makes one believe. I might potentially be interested in redesigning the homepage and improving it a bit aesthetically. My question is, would you be interested in a new homepage design if I design and develop one?

Have a wonderful day :)

### PR DESCRIPTION
**Pull request by yasoob**
Hi! I coded a new website for Dokku. It relies on the latest version of Bootstrap. I have attached some screenshots below for what the website looks like after these changes. Bear with me, there are a lot of screenshots. I provided them all here for the sake of completeness.

Also, there are some nice interactions on the homepage which can't be captured in screenshots. I did make use of `prefers-reduced-motion` media query so that the animations don't run when the user has enabled this setting. I tested it out in Chrome and it works perfectly (i.e disabling the animations).

There is also a subtle gradient animation on the code box on the homepage. It is easy to miss if not paying attention 😄 

I tried to use css variables wherever I could so that the colors are easy to tweak. You can find the main "theme" colors in the `:root` scoped variable list. It is always hard to come up with variable names so pardon me if some names don't make much sense 🤪  I tried my best. Promise! 🤞 

I tried not to touch the css of the documentation page but I did modify it slightly to go with the new theme. 

I hope people like the website proposal ❤️ 

## Homepage light mode (default)
127 0 0 1_5501_home html (1)

## Homepage dark mode
127 0 0 1_5501_home html

## Homepage on mobile screen
127 0 0 1_5501_home html (2)

## Homepage SVG animation
universe animation

## Tooltip on SVG group hover

If you hover over any of the logos in the SVG, the animation pauses and shows you a tooltip containing the name of the software/company. The logos are also clickable
image

## Tooltip on avatar hover
Hovering over an avatar pauses the animation and shows a tooltip with the name of the user and clicking on the image takes you to their homepage as configured on OpenCollective
image

## Shows menu toggled
image

## Documentation page light mode
127 0 0 1_5501_template html (1)

## Documentation page dark mode
127 0 0 1_5501_template html (2)

### CODE REVIEW

File: docs/template.html [hunk 1]
```diff
@@ -41,9 +41,15 @@
...
-    <!-- <link href="/dokku/docs/assets/style.css" rel="stylesheet"> -->
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor" crossorigin="anonymous">
+
+    <!-- Font (Lato) -->
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Lato:wght@400;700;900&display=swap" rel="stylesheet">
+    
+    <!-- <link href="https://cdn.jsdelivr.net/gh/dokku/dokku@v0.27.4/docs/assets/style.css" rel="stylesheet"> -->
+    <link href="./assets/style.css" rel="stylesheet">
```
Comment (josegonzalez) [inline docs/template.html:52]: This needs to be versioned on merge, just fyi.

Comment (yasoob) [reply docs/template.html:52]: Sounds good. I just wasn't sure what to use currently as the latest css isn't on GitHub. I believe I can safely uncomment the line and the CSS would become available once the PR is merged. Right?

**Comment by josegonzalez** [comment]
This is great! A few thoughts:

- I think we need a second, larger set of icons for the "bigger" opencollective sponsors. Any way to work that in?
- Could we switch the svg group hover thing to be on the left vs the right? That'll hopefully balance out the page.
- I'll fix this if need be after merge, but maybe having the actual commands vs commented out in the example would be good?

    ```shell
    # download the installation script
    wget https://dokku.com/installer/latest/bootstrap.sh
    # run the installer
    sudo DOKKU_TAG=v0.27.4 bash bootstrap.sh
    # Configure your server domain
    dokku domains:set-global dokku.me
    # and your ssh key to the dokku user
    echo "your-public-key-contents-here" | dokku ssh-keys:add admin
    # create your first app and you're off to the races!
    dokku apps:create test-app
    ```

File: docs/template.html (hunk 1 above)
Comment (josegonzalez) [reply docs/template.html]: Once we merge it, I'll need to release to get it out. The workflow here is rough for development :(

Comment (josegonzalez) [reply docs/template.html]: Once we merge it, I'll need to release to get it out. The workflow here is rough for development :(

**Comment by yasoob** [comment]
SVG hover group is now to the left
image

Quickstart instructions have been updated
image

Still thinking on your first point. Need to figure out how best to present them.

**Comment by yasoob** [comment]
@josegonzalez I went through the OpenCollective API and I couldn't find a way to get the list of biggest financial contributors. [This is the page I am referring to](https://docs.opencollective.com/help/contributing/development/api/collectives#get-members-per-tier). The issue is that Dokku doesn't have different financial tiers with significant difference. 

If I go with the "Sponsors" tier, it only returns one org. [API url](https://opencollective.com/dokku/members/all.json?TierId=202&limit=50&offset=0)

Thoughts?

Other than this do you see anything as a blocker for this PR?

**Comment by yasoob** [comment]
Also @josegonzalez thoughts on using "An open source PAAS alternative to Heroku." instead of "The smallest PAAS implementation you have ever seen." and maybe replacing the sub-heading with "Dokku is a single-host PAAS that helps you build and manage the lifecycle of applications." on the homepage?

This makes the purpose of Dokku clear and also lists the biggest difference (single-host) in the main heading.

**Comment by josegonzalez** [comment]
It's not actually single-host, you can use it against Kubernetes or Nomad (the plugins are official). The biggest difference is that you self-host vs having someone else manage it for you.

**Comment by yasoob** [comment]
I was under the impression that it is single host after looking at some articles + a Twitter thread. Don't have the links handy but well, in that case, we can make the sub-heading "Dokku helps you build and manage the lifecycle of applications from building to scaling"?

**Comment by josegonzalez** [comment]
Yeah thats probably better :)

**Comment by yasoob** [comment]
Made the copy change.

I feel like the homepage is missing the context that some asciinema can add. This is also something you recommended in the issue #5175 I am already developing the UI. It is 95% done. @josegonzalez do you think you can record three asciinemas for me? The sooner I get them, the quicker I can turn around and finish it off. I am creating three tabs of "Create", "Deploy" and "Scale". Let me know if you have different naming suggestions :)

**Comment by josegonzalez** [comment]
I can do that, most likely wednesday.

**Comment by yasoob** [comment]
Added Asciinema tabs on homepage that will showcase the create, deploy, and ship steps. Latest screenshot attached below.

I went about a slightly round-about way of implementing asciinema because if I embedded the script directly as suggested by asciinema, the player has the height of 0 for the embeds in hidden tabs and so doesn't work when we toggle the hidden tabs.

I observed this behavior only on Chrome. Safari and Firefox worked perfectly fine. But quite a few people use Chrome so need to cater to that as well.

Before we merge this in, we will need to update the asciinema link based on the recordings you provide me.

127 0 0 1_5501_home html (3)

**Comment by yasoob** [comment]
@josegonzalez  did you get a chance to put something together? :)

**Comment by yasoob** [comment]
@josegonzalez really hoping to get this finalized and merged this month if possible. Let me know if I can help other than recording the asciinema :)

**Comment by josegonzalez** [comment]
App creation: https://asciinema.org/a/497945
App deployment: https://asciinema.org/a/497946
App scaling: https://asciinema.org/a/497948

**Comment by josegonzalez** [comment]
These can be turned into gifs like so: https://github.com/asciinema/asciicast2gif

I wonder if we should do that and have a light/dark version of each gif?

**Comment by josegonzalez** [comment]
We can also just embed the asciicasts (would be easier to update them long-term that way).

**Comment by yasoob** [comment]
I just linked the casts. I think the default asciinema color scheme goes well with dark + light mode so there is no need to explicitly create separate color versions. And also, like you said, this way future updates will be easier.

**Comment by yasoob** [comment]
@josegonzalez Just fixed the merge conflicts. The branch is ready for review and merge.  👍

**Comment by yasoob** [comment]
@josegonzalez is there anything I can do to help test this and get it merged? :)

**Comment by josegonzalez** [comment]
Ah sorry. Last week was a bit hectic and I've been recovering. I'll have some time to look at this Wednesday and get it merged :)

**Comment by yasoob** [comment]
Hey @josegonzalez any updates on this? :)

**Comment by yasoob** [comment]
Hi @josegonzalez just checking in. Let me know if I can help in any way to get this done ✅

**Comment by josegonzalez** [comment]
Needs a rebase which I was gonna try to do this weekend and ship it for Monday. If you wanna do that, we'll be good :)

**Comment by yasoob** [comment]
Done :)

**Comment by yasoob** [comment]
It seems like you need to do some release procedure for the website to work correctly. The stylesheet is not the one that I included in this PR and some assets are not loading (e.g the svg logo). This is what I see on my end:
image

**Comment by josegonzalez** [comment]
Yeah I'm doing that now. I triggered a patch release as soon as I clicked the merge button.

**Comment by yasoob** [comment]
Sounds good. You might want to tweak the asset paths a bit because the logo and the social nav bar icons (on the documentation pages) are not loading due to a path issue.

image

image

**Comment by josegonzalez** [comment]
@yasoob mind going through and doing some qa on the site?

**Pull request merged**